packages/trpc/server/routers/viewer/workflows/update.handler.ts

```diff
@@ -39,9 +39,49 @@ type UpdateOptions = {
   input: TUpdateInputSchema;
 };
 
+const verifyEmailSender = async (
+  email: string,
+  userId: number,
+  teamId: number | null,
+  prisma: PrismaClient
+) => {
+  const verifiedEmail = await prisma.verifiedEmail.findFirst({
+    where: {
+      email,
+      OR: [{ userId }, { teamId }],
+    },
+  });
+
+  if (!verifiedEmail) {
+    throw new TRPCError({ code: "NOT_FOUND", message: "Email not verified" });
+  }
+
+  if (teamId) {
+    if (!verifiedEmail.teamId) {
+      await prisma.verifiedEmail.update({
+        where: {
+          id: verifiedEmail.id,
+        },
+        data: {
+          teamId,
+        },
+      });
+    } else if (verifiedEmail.teamId !== teamId) {
+      await prisma.verifiedEmail.create({
```

Comment:
Suppose user has multiple teams and email was verified for User workflow and then was added to Team A, but now user also used same email for Team B, then create a new entry for Team B